ObjectiveGit/GTConfiguration.m

```diff
@@ -58,11 +60,10 @@ - (void)setString:(NSString *)s forKey:(NSString *)key {
 }
 
 - (NSString *)stringForKey:(NSString *)key {
-	const char *string = NULL;
-	git_config_get_string(&string, self.git_config, key.UTF8String);
-	if (string == NULL) return nil;
+	git_buf buffer = {};
+	if (git_config_get_string_buf(&buffer, self.git_config, key.UTF8String) != 0) return nil;
 
-	return [NSString stringWithUTF8String:string];
+	return [[NSString alloc] initWithData:[NSData git_dataWithBuffer:&buffer] encoding:NSUTF8StringEncoding];
```

Comment:
[Collapsed context](https://github.com/libgit2/objective-git/pull/446#discussion_r25971693). We have an `NSData` method for this, but no `NSString` one. I feel adding this to the `NSString` category wouldn’t contribute any significant value, so I’ve just used the `NSData` one.